doc/regexp.rdoc

```diff
@@ -1,753 +1,1171 @@
-    \Z/x
-    float_pat.match('3.14') #=> #<MatchData "3.14" 1:".14">
+- <tt>/[[:ascii:]]/</tt>: Matches a character in the ASCII character set.
+- <tt>/[[:word:]]/</tt>: Matches a character in one of these Unicode character
+  categories (see below):
 
-There are a number of strategies for matching whitespace:
+  - +Mark+ (+M+).
+  - +Letter+ (+L+).
+  - +Number+ (+N+)
+  - <tt>Connector Punctuation</tt> (+Pc+).
 
-* Use a pattern such as <tt>\s</tt> or <tt>\p{Space}</tt>.
-* Use escaped whitespace such as <tt>\ </tt>, i.e. a space preceded by a backslash.
-* Use a character class such as <tt>[ ]</tt>.
+=== Comments
 
-Comments can be included in a non-<tt>x</tt> pattern with the
-<tt>(?#</tt><i>comment</i><tt>)</tt> construct, where <i>comment</i> is
-arbitrary text ignored by the regexp engine.
+A comment may be included in a regexp pattern
+using the <tt>(?#</tt>_comment_<tt>)</tt> construct,
+where _comment_ is a substring that is to be ignored.
+arbitrary text ignored by the regexp engine:
 
-Comments in regexp literals cannot include unescaped terminator
-characters.
+  /foo(?#Ignore me)bar/.match('foobar') # => #<MatchData "foobar">
 
-== Encoding
+The comment may not include an unescaped terminator character.
+
+See also {Extended Mode}[rdoc-ref:regexp.rdoc@Extended+Mode].
+
+== Modes
+
+Each of these modifiers sets a mode for the regexp:
+
+- +i+: <tt>/_pattern_/i</tt> sets
+  {Case-Insensitive Mode}[rdoc-ref:regexp.rdoc@Case-Insensitive+Mode].
+- +m+: <tt>/_pattern_/m</tt> sets
+  {Multiline Mode}[rdoc-ref:regexp.rdoc@Multiline+Mode].
+- +x+: <tt>/_pattern_/x</tt> sets
+  {Extended Mode}[rdoc-ref:regexp.rdoc@Extended+Mode].
+- +o+: <tt>/_pattern_/o</tt> sets
+  {Interpolation Mode}[rdoc-ref:regexp.rdoc@Interpolation+Mode].
+
+Any, all, or none of these may be applied.
+
+Modifiers +i+, +m+, and +x+ may be applied to subexpressions:
+
+- A preceding <tt>(?_modifier_)</tt> turns the mode "on."
+- A preceding <tt>(?-_modifier_)</tt> turns the mode "off."
+
+Example:
+
+  re = /(?i)te(?-i)st/
+  re.match('test') # => #<MatchData "test">
+  re.match('TEst') # => #<MatchData "TEst">
+  re.match('TEST') # => nil
+  re.match('teST') # => nil
+
+\Method Regexp#options returns an integer whose value showing
+the settings for case-insensitivity mode, multiline mode, and extended mode.
+
+=== Case-Insensitive Mode
+
+By default, a regexp is case-sensitive:
+
+  /foo/.match('FOO')  # => nil
+
+Modifier +i+ enables case-insensitive mode:
+
+  /foo/i.match('FOO')
+  # => #<MatchData "FOO">
+
+\Method Regexp#casefold? returns whether the mode is case-insensitive.
+
+=== Multiline Mode
+
+By default, a Ruby regexp is in multiline mode,
+which means that anchors <tt>^</tt> and <tt>$</tt> match, respectively,
+the beginning or end of a line in the target string.
+
+Modifier +m+ therefore does not enable multiline mode -- it's already enabled.
+The modifier does, however, make one small difference:
```

Comment:
I think it would be a good idea to write what the modifier does in words, rather than making the reader try to figure it out from the example.

Reply:
Fixed.